src/core_plugins/kibana/public/management/react/index-pattern/index-pattern-view/index-pattern-view.container.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { IndexPatternView as IndexPatternViewComponent } from './index-pattern-view.component';
+
+import {
+  CustomProps
+} from 'plugins/kibana/management/react/lib/custom_props';
+
+import {
+  TabProps
+} from 'plugins/kibana/management/react/lib/tab_props';
+
+import {
+  refreshFields,
+  deleteIndexPattern,
+  setDefaultIndexPattern,
+  fetchIndexPattern,
+} from 'plugins/kibana/management/react/store/actions';
+
+import {
+  getSelectedIndexId,
+  getSelectedIndexFields,
+  getSelectedIndexTimeFieldName,
+  getSelectedIndexIsDefault,
+} from 'plugins/kibana/management/react/store/reducers';
+
+const IndexPatternView = connect(
+  state => ({
+    id: getSelectedIndexId(state),
+    fields: getSelectedIndexFields(state),
+    timeFieldName: getSelectedIndexTimeFieldName(state),
+    isDefault: getSelectedIndexIsDefault(state),
+  }),
+  { refreshFields, deleteIndexPattern, setDefaultIndexPattern, fetchIndexPattern },
+)(class extends Component {
+  componentWillMount() {
+    this.props.fetchIndexPattern(this.props.pattern);
```

Comment:
I'm using this pattern too - an initial fetch in componentWillMount. https://github.com/elastic/x-pack-kibana/commit/16138ccbd0e84551f56fc920ad665d5022185501#diff-9ebbae0643c0228e0cdc544051695885R24